backend/main/src/Services/InvitationHashService/InvitationHashService.ts

```diff
@@ -0,0 +1,132 @@
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import { Err, Ok, Result } from "result-ts-type";
+import { IGroupRepository } from "src/Group/Domain/IGroupRepository";
+import { GroupId } from "src/Group/Domain/Entities/Group";
+import { UserId } from "src/User";
+import Md5 from "crypto-js/md5";
+import { DomainObjectError, ServiceError } from "src/Shared/Utils/Errors";
+
+interface IHash {
+	hash: string;
+	expiryDatetime: Date;
+}
+
+export interface IGenerateAnInvitationHashService {
+	generateAnInvitationHash(input: { groupId: string; userId: string }): Promise<
+		Result<IHash, DomainObjectError | GroupNotFound | PermissionError>
+	>;
+}
+
+export interface IJoinToGroupUsingAnInvitationHashService {
+	joinToGroupUsingAnInvitationHash(invitationHash: string): Promise<string>;
+}
+
+export class InvitationHashService implements IGenerateAnInvitationHashService {
+	private nishikiDynamoDBClient: NishikiDynamoDBClient;
+	private groupRepository: IGroupRepository;
+
+	constructor(
+		nishikiDynamoDBClient: NishikiDynamoDBClient,
+		groupRepository: IGroupRepository,
+	) {
+		this.nishikiDynamoDBClient = nishikiDynamoDBClient;
+		this.groupRepository = groupRepository;
+	}
+
+	/**
+	 * This method generate an invitation hash.
+	 * Before generating an invitation hash, this method check existence of the invitation hash.
+	 * If the invitation hash does exist, this method checks the expiry datetime.
+	 * If the expiry datetime is over, this method generate a new invitation hash.
+	 * If not over, this method returns the invitation hash with updating the expiry datetime.
+	 * The invitation hash link is generated by a string of "groupId + expiry Datetime" using MD5.
+	 * https://genesis-tech-tribe.github.io/nishiki-documents/proect-document/specifications/detail/invitation-to-group#invitation-link-specification
```

Comment:
The link doesn't work.
```suggestion
	 * https://genesis-tech-tribe.github.io/nishiki-documents/project-document/specifications/detail/invitation-to-group#invitation-link-specification

```

Reply:
That's true.
I have no idea how I came to this situation and you noticed this one. Incredible.